src/packet/ipv4.rs

```diff
@@ -39,22 +41,43 @@ pub struct Ipv4 {
     payload: Vec<u8>,
 }
 
+
+impl <'p> HasPseudoheader for Ipv4Packet <'p> {
+    fn pseudoheader_checksum(&self) -> u32 {
```

Comment:
I'd like to see some benchmarks for old-style vs new style. Theoretically they should be optimised to the same thing based on my quick look over, but I would rather be sure.